### PR TITLE
Vet clinic database: add "join table" for visits

### PR DESCRIPTION
**Pull request by Eccb7**
## Description

This pull request extends the database for the Vet Clinic project with the addition of new tables and the fulfillment of several project requirements. The following tasks have been completed:

- Created the `vets` table to store information about veterinarians, including their name, age, and date of graduation.
- Implemented a many-to-many relationship between the `species` and `vets` tables using the `specializations` join table.
- Created a many-to-many relationship between the `animals` and `vets` tables using the `visits` join table, which also tracks the date of each visit.
- Inserted data for veterinarians, their specializations, and animal visits into their respective tables.
- Implemented SQL queries to answer various questions about the data, including animal visits, specialties, and more.

### Changes Made

- Added SQL code to create the `vets`, `specializations`, and `visits` tables in the `schema.sql` file.
- Inserted data for veterinarians, specializations, and visits into the `data.sql` file.
- Created SQL queries to answer specific questions about the database in the `queries.sql` file.

## Screenshots

- Create vets, specializations and visits tables.
Screenshot from 2023-09-01 17-06-06

- Insert data into tables
Screenshot from 2023-09-01 17-10-21

- Query results
- Who was the last animal seen by William Tatcher?

Screenshot from 2023-09-01 17-57-28

Screenshot from 2023-09-01 17-13-34

- List all vets and their specialties, including vets with no specialties.

Screenshot from 2023-09-01 17-14-01

- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.

Screenshot from 2023-09-01 17-14-26

- What animal has the most visits to vets?

Screenshot from 2023-09-01 17-14-51

- Who was Maisy Smith's first visit?

Screenshot from 2023-09-01 17-15-11

- Details for the most recent visit: animal information, vet information, and date of visit.

Screenshot from 2023-09-01 17-15-29

- How many visits were with a vet that did not specialize in that animal's species?

Screenshot from 2023-09-01 17-15-46

- What specialty should Maisy Smith consider getting? Look for the species she gets the most.

Screenshot from 2023-09-01 17-16-10

## Checklist

- [x] Created `vets`, `specializations`, and `visits` tables in `schema.sql`.
- [x] Added SQL code for data insertion in `data.sql`.
- [x] Implemented SQL queries for various data questions in `queries.sql`.
- [x] Verified query results.
- [x] Added query result screenshots.

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -101,3 +101,75 @@ LEFT JOIN animals AS a ON o.id = a.owner_id
...
 ORDER BY animal_count DESC
 LIMIT 1;
+
+-- Who was the last animal seen by William Tatcher?
+SELECT a.name AS animal_name
+FROM visits v
+JOIN animals a ON v.animal_id = a.id
+WHERE v.vet_id = 1
+ORDER BY v.visit_date DESC
+LIMIT 1;
```
Comment (tugsem) [inline queries.sql:111]: - [ ] Please add a screenshot for this query result as per the requirement. I suggest you label each of the queries for the next reviewer to find it easily.

**Review by tugsem — CHANGES_REQUESTED** [review]
##  STATUS: REQUIRED CHANGES ♻️
Hello @Eccb7  👋 , 

Good job so far! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

Alt Text
## Highlights 🚀 
✔️screenshots added
✔️query results are correct
✔️required data and tables created 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 1 above)
Comment (Eccb7) [reply queries.sql:111]: Hi @tugsem 
It's included

**Review by LionRouge1 — APPROVED** [review]
Hi @Eccb7, 

# STATUS: APPROVED! 🥇 💯 
Great work on making the changes 👏 
You've done well implementing the requested changes

done

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:

## Highlights
- Great job, your project is error free :+1: 
- Nice work you meet all the requirements :1st_place_medal: 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**